flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/AutoClosableProcess.java

```diff
@@ -167,6 +186,18 @@ private static void processStream(final InputStream stream, final Consumer<Strin
 		).start();
 	}
 
+	private static void processOutputStream(final OutputStream stream, final List<String> inputs) {
+		new Thread(() -> {
```

Comment:
This keeps align with `processInputStream`, we shouldn't block here, because the blocking/non-blocking is determined by user `AutoClosableProcessBuilder#runNonBlocking` or `AutoClosableProcessBuilder#runBlocking()`.

Reply:
I don't think it should be same to `processInputStream`.
For a process, maybe this thread not start, but process ended. I'd like to think of `Input lines` as commands.

Reply:
Even if we don't start a thread here, it is still possible the process is finished when we want to print lines to the process.